src/UI/Areas/Identity/Pages/Account/LoginWith2fa.cshtml

```diff
@@ -33,7 +33,7 @@
 </div>
 <p>
     Don't have access to your authenticator device? You can
-    <a asp-page="./LoginWithRecoveryCode" asp-route-returnUrl="@Model.ReturnUrl">log in with a recovery code</a>.
+    <a id="recovery-code-login" asp-page="./LoginWithRecoveryCode" asp-route-returnUrl="@Model.ReturnUrl">log in with a recovery code</a>.
```

Comment:
Interesting - I assume all these id's are added for testability so you can find the elements? Seems a'ight to me.